### PR TITLE
Add and remove

### PR DESCRIPTION
**Pull request by mutinhiri**
In this Pr I configured Redux store and write actions an=d reducers for adding and removing books.

-Added UI elements to components and make them dispatch actions

### CODE REVIEW

File: src/components/BookList.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+          booksStore.map((book) => (
+            <Book key={book.id} book={book} />
+          ))
+        }
+      </ul>
+      <form action="" onSubmit={submitBookToStore}>
+        <input type="text" name="" placeholder="Book" id="books-input" required />
+        <input type="text" name="" placeholder="Author" id="author-input" required />
+        <button type="submit">Add Book</button>
+      </form>
```
Comment (TedLivist) [inline src/components/BookList.js]: - [ ] In keeping with the SOLID principle, kindly extract this new book form and its other functionalities into a separate component.
- [x] Your inputs are not controlled. Your inputs should be controlled and states should be the only source of truth. You can refer to the first paragraph of this [article](https://reactjs.org/docs/forms.html) to get an idea of what this means. As listed in the project requirements (see image below) your add book form component should have a local `state` for `title` and `author`. And that local state should be updated whenever a user updates the title and author inputs with `onchange`. You can refer to this [article](https://dmitripavlutin.com/controlled-inputs-using-react-hooks/) for a quick solution on how to create controlled components with React hooks.

  muth-req

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @mutinhiri,

Good job so far!
✔️ Good documentation
✔️ Clean code
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/App.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import {
+  BrowserRouter as Router,
+  Routes,
+  Route,
+} from 'react-router-dom';
+import Header from './components/Header';
+import BooksList from './components/BookList';
+import './App.css';
+
+function App() {
```
Comment (TedLivist) [inline src/App.js]: - [x] Let's use ES6 syntax (arrow function) for our functions.

File: src/components/Book.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+import { removeBook } from '../redux/books/books';
+
+function Book(props) {
```
Comment (TedLivist) [inline src/components/Book.js]: - [x] Kindly use ES6 syntax for your functions

File: src/components/BookList.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
+import { v4 as uuidv4 } from 'uuid';
+import { useSelector, useDispatch } from 'react-redux';
+// import React from 'react';
+import { addBook } from '../redux/books/books';
+import Book from './Book';
+
+function BooksList() {
```
Comment (TedLivist) [inline src/components/BookList.js]: - [x] Use ES6 syntax

File: src/components/Header.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+
+export default function Header() {
```
Comment (TedLivist) [inline src/components/Header.js]: - [x] Kindly use ES6 syntax

**Review by Ghiftee — APPROVED** [review]
## STATUS: APPROVED ♻️♻️
Hi @mutinhiri , 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀  
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

**Comment by mutinhiri** [comment]
thank you @Ghiftee

**Pull request merged**